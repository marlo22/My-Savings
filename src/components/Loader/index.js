import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import styled, {ThemeContext } from 'styled-components';

import { Spinner } from 'native-base';

const WrapperView = styled.View`
  height: 100%;
  justify-content: center;
  align-items: center;
`;

const MessageText = styled.Text`
  color: ${props => props.theme.fontColor}
  text-align: center;
`;

const Loader = ({ message }) => {
  const themeContext = useContext(ThemeContext);

  return (
    <WrapperView>
      <Spinner color={themeContext.primaryColor} />
      <MessageText>
        {message}
      </MessageText>
    </WrapperView>
  );
}

Loader.propTypes = {
  message: PropTypes.string.isRequired
};

export default Loader;