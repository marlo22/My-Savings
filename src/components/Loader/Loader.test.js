import React from 'react';
import { mount } from 'enzyme';
import Loader from './';
import { ThemeContext } from 'styled-components/native';

const message = 'Lorem ipsum';
const primaryColor = '#ff0000';
const fontColor = '#000000';

const component = mount(
  <ThemeContext.Provider value={{ primaryColor, fontColor }}>
    <Loader message={message} />
  </ThemeContext.Provider>
);

describe('<Loader />', () => {
  test('should render without errors', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should set primary color as Spinner color', () => {
    const spinner = component.find('Spinner');
    expect(spinner.prop('color')).toBe(primaryColor);
  });

  test('should set primary color as Spinner color', () => {
    const messageText = component.find('Text').last();
    expect(messageText.text()).toBe(message);
  });
});