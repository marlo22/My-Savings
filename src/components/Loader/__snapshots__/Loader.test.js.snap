// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> snapshot 1`] = `
<Loader
  message="Lorem ipsum"
>
  <Styled(View)>
    <StyledNativeComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(View)",
          "inlineStyle": InlineStyle {
            "rules": Array [
              "
  height: 100%;
  justify-content: center;
  align-items: center;
",
            ],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "height": "100%",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Styled(Spinner)
            color="#ff0000"
          >
            <Spinner
              color="#ff0000"
              style={
                Object {
                  "height": 80,
                }
              }
            >
              <ActivityIndicator
                animating={true}
                color="#ff0000"
                hidesWhenStopped={true}
                size="large"
                style={
                  Object {
                    "height": 80,
                  }
                }
              >
                <ActivityIndicator
                  animating={true}
                  color="#ff0000"
                  hidesWhenStopped={true}
                  size="large"
                  style={
                    Object {
                      "height": 80,
                    }
                  }
                />
              </ActivityIndicator>
            </Spinner>
          </Styled(Spinner)>
          <Styled(Text)>
            <StyledNativeComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "displayName": "Styled(Text)",
                  "inlineStyle": InlineStyle {
                    "rules": Array [
                      "
  color: ",
                      [Function],
                      "
  text-align: center;
",
                    ],
                  },
                  "propTypes": Object {
                    "accessible": [Function],
                    "adjustsFontSizeToFit": [Function],
                    "allowFontScaling": [Function],
                    "disabled": [Function],
                    "ellipsizeMode": [Function],
                    "maxFontSizeMultiplier": [Function],
                    "minimumFontScale": [Function],
                    "nativeID": [Function],
                    "numberOfLines": [Function],
                    "onLayout": [Function],
                    "onLongPress": [Function],
                    "onPress": [Function],
                    "pressRetentionOffset": [Function],
                    "selectable": [Function],
                    "selectionColor": [Function],
                    "style": [Function],
                    "suppressHighlighting": [Function],
                    "testID": [Function],
                    "textBreakStrategy": [Function],
                  },
                  "render": [Function],
                  "styledComponentId": "StyledNativeComponent",
                  "target": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "textAlign": "center",
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Lorem ipsum
                </Text>
              </Text>
            </StyledNativeComponent>
          </Styled(Text)>
        </View>
      </View>
    </StyledNativeComponent>
  </Styled(View)>
</Loader>
`;
